### PR TITLE
Reconcile should log error if a resource is not found in detect Bindable resource

### PR DESCRIPTION
**Pull request by Avni-Sharma**
Fix for #584 
Do not error out when resource not found in the binding resources, Instead log it as stated below
```
2020-07-23T21:04:33.309+0530	DEBUG	reconciler.buildServiceContexts	Resource not found in Bindable Resources	{"Request.Namespace": "try", "Request.Name": "binding-request-etcd", "ServiceBindingRequest.Name": "binding-request-etcd", "Error": "the server could not find the requested resource"}
```

### CODE REVIEW

File: pkg/controller/servicebindingrequest/service.go [hunk 1]
```diff
@@ -170,9 +173,14 @@ func getOwnedResources(
 	error,
 ) {
 	var resources []*unstructured.Unstructured
+BINDING_RESOURCES:
```
Comment (sbose78) [inline pkg/controller/servicebindingrequest/service.go:176]: This is stray text 🤔

Comment (Avni-Sharma) [reply pkg/controller/servicebindingrequest/service.go:176]: this is a label used in L182 with continue for better clarity

**Comment by dharmit** [comment]
Works for me on minikube. I'm able to create a deployment (`odo push`) and link it with a running service (`odo link`).

File: pkg/controller/servicebindingrequest/service.go (hunk 1 above)
Comment (baijum) [reply pkg/controller/servicebindingrequest/service.go:176]: I think `continue` without the label is not confusing.

Comment (DhritiShikhar) [reply pkg/controller/servicebindingrequest/service.go:176]: @baijum Looks okay to me. Are you okay with this change?

Comment (baijum) [reply pkg/controller/servicebindingrequest/service.go:176]: @DhritiShikhar I think the label can be removed.

Comment (Avni-Sharma) [reply pkg/controller/servicebindingrequest/service.go:176]: was trying to stay consistent with https://github.com/redhat-developer/service-binding-operator/blob/master/pkg/controller/servicebindingrequest/servicecontext.go#L62

**Pull request merged**